x/oracle/abci.go

```diff
@@ -106,9 +113,18 @@ func CalcPrices(ctx sdk.Context, params types.Params, k keeper.Keeper) error {
 		}
 	}
 
+	// Get validators that can earn rewards in this Slash Window
+	validatorRewardSet := types.ValidatorRewardSet{
+		ValidatorMap: make(map[string]bool),
+	}
+	k.CurrentValidatorRewardSet(ctx, func(valRewardSet types.ValidatorRewardSet) bool {
+		validatorRewardSet = valRewardSet
+		return false
+	})
+
 	// update miss counting & slashing
 	voteTargetsLen := len(params.MandatoryList)
-	claimSlice := types.ClaimMapToSlice(validatorClaimMap)
+	claimSlice := types.ClaimMapToSlice(validatorClaimMap, validatorRewardSet.ValidatorMap)
```

Comment:
Correct me if I'm wrong, but if `claimSlice` only contains validators which are eligible for rewards, I don't think it'll update the miss counter for ineligible validators:

```golang
	for _, claim := range claimSlice {
		misses := uint64(voteTargetsLen - int(claim.MandatoryWinCount))
		if misses == 0 {
			continue
		}

		// Increase miss counter
		k.SetMissCounter(ctx, claim.Recipient, k.GetMissCounter(ctx, claim.Recipient)+misses)
	}
```

we still want to be able to jail underperforming bad validators